mathics/core/number.py

```diff
@@ -56,7 +56,12 @@ def _get_float_inf(value, evaluation) -> typing.Optional[float]:
     return value.round_to_float(evaluation)
 
 
-def get_precision(value, evaluation) -> typing.Optional[int]:
+def get_precision(value, evaluation, show_messages=True) -> typing.Optional[int]:
```

Comment:
If this returns float, then the signature `Optional[int]` needs adjusting.

Reply:
You are right! I just left it as I found it. What I am not sure about is if we have to change the signature or the returned value...

Reply:
Line 67, the assignment to the return value is:   `d = value.round_to_float(evaluation)`

Reply:
OK, but the function, should return an integer or a floating-point number?

Reply:
Good point! The other places where `get_precision()` appears it returns an int. And semantically it should be an `int`. 

We don' thave a `value.round_to_int(evaluation)` so I believe that should be added and used here.

Reply:
Looking at it more carefully, it seems that for compatibility with WMA, it should be a Float: WMA uses real values for precision, to take into account the internal representation of numbers. This is why $MachinePrecision is not 16, but 15.9546`

Reply:
Again - good work! 

What I'd ask though is emphasize that this is a float and not an int. If nothing else it will prevent future people inspecting the code from taking the time to wonder about it as we just did.

And then this calls into question whether the other corresponding `get_precision()` methods shouldn't be changed to return a float as well.